pkg/operator/csi/csidrivercontrollerservicecontroller/helpers.go

```diff
@@ -190,8 +190,43 @@ func WithPlaceholdersHook(configInformer configinformers.SharedInformerFactory)
 		logLevel := loglevel.LogLevelToVerbosity(spec.LogLevel)
 		pairs = append(pairs, []string{"${LOG_LEVEL}", strconv.Itoa(logLevel)}...)
 
-		// Default TLS Cipher Suites for Kube RBAC sidecars
-		pairs = append(pairs, []string{"${TLS_CIPHER_SUITES}", defaultTLSCipherSuites}...)
+		replaced := strings.NewReplacer(pairs...).Replace(string(manifest))
+		return []byte(replaced), nil
+	}
+}
+
+// WithServingInfo is a manifest hook that replaces ${TLS_CIPHER_SUITES} and ${TLS_MIN_VERSION}
+// placeholders with the observed configuration.
+func WithServingInfo() dc.ManifestHookFunc {
+	return func(opSpec *opv1.OperatorSpec, manifest []byte) ([]byte, error) {
+		if len(opSpec.ObservedConfig.Raw) == 0 {
+			return manifest, nil
+		}
+
+		var config map[string]interface{}
+		if err := json.Unmarshal(opSpec.ObservedConfig.Raw, &config); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal the observedConfig: %w", err)
+		}
+
+		cipherSuites, cipherSuitesFound, err := unstructured.NestedStringSlice(config, csiconfigobservercontroller.CipherSuitesPath()...)
+		if err != nil {
+			return nil, fmt.Errorf("couldn't get the servingInfo.cipherSuites config from observed config: %w", err)
+		}
+
+		minTLSVersion, minTLSVersionFound, err := unstructured.NestedString(config, csiconfigobservercontroller.MinTLSVersionPath()...)
+		if err != nil {
+			return nil, fmt.Errorf("couldn't get the servingInfo.minTLSVersion config from observed config: %v", err)
+		}
+
+		pairs := []string{}
+		if cipherSuitesFound && len(cipherSuites) > 0 {
+			pairs = append(pairs, []string{"${TLS_CIPHER_SUITES}", strings.Join(cipherSuites, ",")}...)
+
+		}
+
+		if minTLSVersionFound && len(minTLSVersion) > 0 {
```

Comment:
This hook will produce invalid Deployments (with literally `--tls-cipher-suites=${TLS_CIPHER_SUITES}`) until the observer finishes the initial `APIServer` CR observation. Is that what we want? Maybe `--tls-min-version=""  --tls-cipher-suites=""` would be a better default.

Reply:
Can the --tls-* arguments be omitted in that case? I'm not sure what behavior to expect if we set those arguments to empty strings, but if we omit those args, the deployment will be valid and kube-rbac-proxy will select its own defaults until the observer returns the value from APIServer.

Reply:
> Can the --tls-* arguments be omitted in that case

It's actually hard, right now do a simple search for `${TLS_CIPHER_SUITES}` string and replace it. We would need to change that `${TLS_CIPHER_SUITES}` to contain also `--tls-cipher-suites` (which is easy), but then the kube-rbac-proxy needs to be able to handle empty string arguments.

Reply:
We shouldn't let any values other than those specified in the TLS profile to be set. For instance, if those args are set to empty strings, insecure ciphers would be specified instead (as per Go default ciphers). This is the worse possible outcome.

IMO the current behavior is fine, it's preferable to fail to start and it would only produce an invalid Deployment during cluster installation. I did add an additional check to return a different error message, though.